tracer/src/Datadog.Trace/Iast/TaintedObjects.cs

```diff
@@ -9,19 +9,50 @@ namespace Datadog.Trace.Iast
 {
     internal class TaintedObjects
     {
+        private static readonly bool _largeNumericCache = false;
         private readonly ITaintedMap _map;
 
+        static TaintedObjects()
+        {
+            // From Net 8.0 onwards first 300 digit strings are cached instead of first 10
+            _largeNumericCache = object.ReferenceEquals(299.ToString(), 299.ToString());
+        }
+
         public TaintedObjects()
         {
             _map = new DefaultTaintedMap();
         }
 
-        public void TaintInputString(string stringToTaint, Source source)
+        public bool TaintInputString(string stringToTaint, Source source)
         {
-            if (!string.IsNullOrEmpty(stringToTaint))
+            if (!IsFiltered(stringToTaint))
             {
                 _map.Put(new TaintedObject(stringToTaint, IastUtils.GetRangesForString(stringToTaint, source)));
+                return true;
+            }
+
+            bool IsFiltered(string arg)
+            {
+                // Try to bail out ASAP
+                if (arg.Length == 0) { return true; }
+
+                // 0 - 9 are cached only
+                if (!_largeNumericCache)
+                {
+                    if (arg.Length > 1 || !char.IsDigit(arg[0])) { return false; }
+                    return true;
```

Comment:
I think this is easier to read 😃 
```suggestion
                    return arg.Length == 1 && char.IsDigit(arg[0]);
```